duckduckgo/Android
# Increase leaderboard count

Task/Issue URL: https://app.asana.com/0/488551667048375/525633294284205
Tech Design URL: N/A

**Description**:
This task updates the preexisting logic to wait until at least 30 sites are visited before displaying the tracker networks leaderboard on the Privacy Dashboard.

Note this was not quite the 1-liner expected. It introduces a new db table to support counting the total number of unique sites visited. Some lint tidy ups were also performed.

**Steps to test this PR**:
1. Visit 30 sites, preferably sites with lots of trackers (you may want to edit `LEADERNOARD_MIN_DOMAINS` in `PrivacyDashboardViewModel.kt` to reduce this to 10)
2. Open the dashboard
3. Ensure the leaderboard is displayed

**Migration test:**
Install this PR over a preexisting version of the app (whatever is in develop) to ensure the db migration succeeds.

****


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
